lib/src/authorization_code_grant.dart

```diff
@@ -87,18 +87,18 @@ class AuthorizationCodeGrant {
   final String _delimiter;
 
   /// The HTTP client used to make HTTP requests.
-  http.Client _httpClient;
+  http.Client? _httpClient;
```

Comment:
This shouldn't be nullable

Reply:
I agree, however we currently use null to signal that it's been closed: https://github.com/dart-lang/oauth2/blob/95b6c8d96dc37a1723480961665d3477a46dd303/lib/src/client.dart#L185